src/server/database/database_updater.py

```diff
@@ -0,0 +1,54 @@
+from datetime import date, timedelta
+from typing import Optional
+
+import requests
+
+from .. import QUERIES
+from . import DatabaseProvider
+
+
+class DatabaseUpdater:
+    """Class for updating the database with new stock matket data."""
+
+    @staticmethod
+    def daily_database_update() -> None:
+        """Update the database with prices up to the current day."""
+        today_date: date = date.today()
+        last_known_date: date = DatabaseUpdater.check_last_known_date()
+        print(f"DEBUG: Daily database update triggered. Today is {today_date}.")
+
+        if today_date == last_known_date:
+            print("DEBUG: Nothing to update.")
+
+        while last_known_date < today_date:
+            current_date: date = last_known_date + timedelta(days=1)
+            print(f"DEBUG: Updating price for {current_date}.")
+            DatabaseUpdater.update_selected_date(current_date)
+            last_known_date = DatabaseUpdater.check_last_known_date()
+
+    @staticmethod
+    def check_last_known_date() -> date:
+        """Check the date of last known price."""
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_LAST_KNOWN_DATE)
+            last_known_date: date = handler().fetchall()[0][0].date()
+            print(last_known_date)
+            return last_known_date
+
+    @staticmethod
+    def update_selected_date(selected_date: date) -> None:
+        """Fetch price for chosen date and put it in the database."""
+        date_string_dmy: str = selected_date.strftime("%d-%m-%Y")
+        url = f"https://api.coingecko.com/api/v3/coins/bitcoin/history?date={date_string_dmy}"
+        response = requests.get(url)
+        price: float = float(response.json()["market_data"]["current_price"]["usd"])
+
+        date_string_ymd: str = selected_date.strftime("%Y-%m-%d")
+        with DatabaseProvider.handler() as handler:
+            handler().execute(
+                QUERIES.INSERT_PRICE,
+                (
+                    date_string_ymd,
+                    price,
+                ),
+            )
```

Comment:
Can this method return a boolean whether the insertion was successful or not? It would improve the loop in `daily_database_update()`. Right now the flow is:
1. Check in a database for the first missing date;
2. Insert into the database a missing exchange rate;
3. Update the first missing date by again checking in the database.

The last step can be omitted by setting `last_known_date` to `current_date` if an insertion was a success.
There is a possibility that we will miss a date insertion if it fails enough times, but that is already set in stone by the SQL query.
At the end of the day, I don't think anyone will care about it, but we could at least remove the additional check in the mentioned loop.